sips/pending/_posts/2016-06-25-trailing-commas.md

```diff
@@ -0,0 +1,146 @@
+
+{% highlight scala %}
+val xs = Seq(
+  foo,
+  baz   // This isn't going to work
+  bar,
+)
+{% endhighlight %}
+
+### Reduce diff noise
+
+Allowing trailing commas also reduces a lot of noise in diffs, such as:
+
+{% highlight diff %}
+@@ -4,7 +4,8 @@
+ Map(
+   foo,
+   bar,
+-  baz
++  baz,
++  quux
+ )
+{% endhighlight %}
+
+### VCS authorship attribution
+
+Using the example above, the authorship of the `baz` line would be preserved, instead of becoming that of the author of the `quux` line.
+
+### Simplify code generation
+
+Such a feature would also simplify generating Scala source code.
+
+### Long standing ticket
+
+There is an open ticket ([SI-4986][]) where this feature was requested, referencing the fact that it facilitates code generation by tools and allows for easier sorting of the values, initially in the context of import selectors but later also for other constructs in the syntax.
+
+### Real-world use-cases
+
+Some real-world use-cases where elements of a sequence are typically added, removed or moved are:
+
+* invoking constructors or methods (such as `apply` or `copy`) which present a lot of options defined with default values
+* `settings(...)` arguments or elements of `libraryDependencies`, `scalacOptions` or `javaOptions` sequences in sbt
+
+## Design Decisions
+
+There are a number of different elements of the Scala syntax that are comma separated, but instead of changing them all a subset of the more useful ones was chosen:
+
+* tuples
+* argument and parameter groups, including for implicits, for functions, methods and constructors
+* import selectors
+
+From the spec these are:
+
+* SimpleExpr1, ArgumentExprs via Exprs
+* ParamClause, ParamClauses via Params
+* ClassParamClause, ClassParamClauses via ClassParams
+* ImportSelector
+
+The elements that have not changed are:
+
+* ValDcl, VarDcl, VarDef via ids
+* Type via FunctionArgTypes
+* SimpleType, TypeArgs via Types
+* Expr, ResultExpr via Bindings
+* SimplePattern via Patterns
+* TypeParamClause, FunTypeParamClause
+* ImportExp
+* PatDef
+
+## Implementation
+
+The implementation is a simple change to the parser, allowing for a trailing comma, for the groups detailed above, and has been proposed in [scala/scala#5245][].
+
+## Drawbacks/Trade-offs
+
+The drawback, or trade-off, to this change is that it adds another way in which it is possible to do something in Scala. But it is the opinion of this SIP that the pragmatic advantage of being able to have trailing commas is worth this drawback.
+
+Given that this is a change in syntax, another drawback is that it requires changing the existing tools, such as those that parse Scala: intellij-scala, scalariform, scala.meta and scalaparse.
+
+Another drawback is that for projects that cross build to previous versions of Scala they would have to take into account that this feature wouldn't be available for those versions (assuming this feature isn't backported).
```

Comment:
@sjrd I specifically mentioned Cross Build Hinderance (https://github.com/scala/scala.github.com/pull/533#issuecomment-239281742) as a drawback.

It's a drawback worth considering for this SIP in particular because the added benefits are not as feature-based, and closer to the pro/con 50:50 point than, let's say SIP-14 Futures and Promises. Sure adding Future to the standard library would prevent cross building, but it does more things.

Another reason to think about this specifically for this SIP is that the surface area that might affect this SIP is potentially all Scala sources, which I don't think can be said of just adding a method to the standard library. Closer analogy might be SIP-18 Modularizing Language Features which might have added `import language.implicitConversions` etc many existing source. The potential surface area becomes relevant especially given the rise of auto formatting of Scala source using Scalariform or Scalafmt.